src/main/java/org/tikv/common/operation/RegionErrorHandler.java

```diff
@@ -264,8 +269,34 @@ public TiRegion getRegion() {
     return recv.getRegion();
   }
 
-  private void invalidateRegionStoreCache(TiRegion ctxRegion) {
+  private void invalidateRegionStoreCache(TiRegion ctxRegion, long storeId) {
     regionManager.invalidateRegion(ctxRegion);
-    regionManager.invalidateStore(ctxRegion.getLeader().getStoreId());
+    regionManager.invalidateStore(storeId);
+    notifyRegionStoreCacheInvalidate(
+        ctxRegion.getId(), storeId, CacheInvalidateEvent.CacheType.REGION_STORE);
+  }
+
+  private void notifyRegionStoreCacheInvalidate(
```

Comment:
It's better to figure out whether we can asynchronizedly run callbacks.
Since these callbacks may cost a lot of time

Reply:
Add a new thread to run